bson/raw_bson.py

```diff
@@ -13,6 +13,62 @@
 # limitations under the License.
 
 """Tools for representing raw BSON documents.
+
+Inserting and Retrieving RawBSONDocuments
+=========================================
+
+Example: Moving a document between different databases/collections
+
+.. testsetup::
+  from pymongo import MongoClient
+  client = MongoClient("localhost", 27017, document_class=RawBSONDocument)
+  client.drop_database('db')
+  client.drop_database('replica_db')
+
+.. doctest::
+
+  >>> import bson
+  >>> from pymongo import MongoClient
+  >>> from bson.raw_bson import RawBSONDocument
+  >>> client = MongoClient("localhost", 27017, document_class=bson.)
+  >>> db = client.db
+  >>> result = db.test.insert_many([
+  ...  {'a': 1},
+  ...  {'b': 1},
+  ...  {'c': 1},
+  ...  {'d': 1}])
+  >>> assert result.acknowledged
+  >>> replica_db = client.replica_db
+  >>> for doc in db.test.find():
+  ...    print(f"raw document: {doc.raw}")
+  ...    print(f"decoded document: {bson.decode(doc.raw)}")
+  ...    result = replica_db.test.insert_one(doc)
+  ...    assert result.acknowledged
+  raw document: b'...'
+  decoded document: {'_id': ObjectId('...'), 'a': 1}
+  raw document: b'...'
+  decoded document: {'_id': ObjectId('...'), 'b': 1}
+  raw document: b'...'
+  decoded document: {'_id': ObjectId('...'), 'c': 1}
+  raw document: b'...'
+  decoded document: {'_id': ObjectId('...'), 'd': 1}
+  >>> for doc in replica_db.test.find():
```

Comment:
How much value does iterating the `replica_db.test` collection in this example provide?

Reply:
It was mostly just to show that the documents in the new collection match the ones in the old one. I'm not sure how I'd change it aside from just straight removing it (which I'd be ok with doing). Let me know if you have any suggestions!

Reply:
I'd be fine with removing this entirely. @ShaneHarvey can make the final call.

Reply:
Let's remove it.